drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -194,84 +255,93 @@ static void gps_thread(int dev_ptr)
 				LOG_DBG("Waiting for time window to operate");
 
 				operation_blocked = true;
-				continue;
-			}
+				evt.type = GPS_EVT_OPERATION_BLOCKED;
 
-			operation_blocked = false;
+				notify_event(dev, &evt);
 
-			if (pvt_deadline_missed(&raw_gps_data.pvt)) {
-				LOG_DBG("Invalid PVT frame, discarding");
 				continue;
-			}
+			} else if (operation_blocked) {
+				/* GPS has been unblocked. */
+				LOG_DBG("GPS has time window to operate");
 
-			print_satellite_stats(&raw_gps_data);
-			copy_pvt(&fresh_pvt.pvt, &raw_gps_data.pvt);
+				operation_blocked = false;
+				evt.type = GPS_EVT_OPERATION_UNBLOCKED;
 
-			if ((drv_data->trigger.chan == GPS_CHAN_PVT)
-			   && (drv_data->trigger.type == GPS_TRIG_DATA_READY)) {
-				trigger_send = true;
-				LOG_DBG("PVT data ready");
+				notify_event(dev, &evt);
 			}
 
-			if ((drv_data->trigger.type == GPS_TRIG_FIX) &&
-			    is_fix(&fresh_pvt.pvt)) {
-				if (drv_data->trigger.chan == GPS_CHAN_PVT) {
-					trigger_send = true;
-				}
+			copy_pvt(&evt.pvt, &raw_gps_data.pvt);
+
+			if (is_fix(&raw_gps_data.pvt)) {
 				LOG_DBG("PVT: Position fix");
+
+				evt.type = GPS_EVT_PVT_FIX;
 				fix_timestamp = k_uptime_get();
+				has_fix = true;
+				on_fix(dev);
+			} else {
+				evt.type = GPS_EVT_PVT;
 			}
 
-			break;
+			notify_event(dev, &evt);
+			print_satellite_stats(&raw_gps_data);
 
+			break;
 		case NRF_GNSS_NMEA_DATA_ID:
 			if (operation_blocked) {
 				continue;
 			}
 
-			memcpy(fresh_nmea.nmea.buf, raw_gps_data.nmea, len);
-			fresh_nmea.nmea.len = strlen(raw_gps_data.nmea);
+			memcpy(evt.nmea.buf, raw_gps_data.nmea, len);
 
-			if ((drv_data->trigger.chan == GPS_CHAN_NMEA) &&
-			    (drv_data->trigger.type == GPS_TRIG_DATA_READY)) {
-				trigger_send = true;
-				LOG_DBG("NMEA data ready");
-			}
+			/* Don't count null temrinator. */
+			evt.nmea.len = len - 1;
 
-			if ((drv_data->trigger.type == GPS_TRIG_FIX)
-			    && is_fix(&fresh_pvt.pvt)) {
-				if (drv_data->trigger.chan == GPS_CHAN_NMEA) {
-					trigger_send = true;
-				}
+			if (has_fix) {
 				LOG_DBG("NMEA: Position fix");
+
+				evt.type = GPS_EVT_NMEA_FIX;
+			} else {
+				evt.type = GPS_EVT_NMEA;
 			}
-			break;
 
-		default:
+			notify_event(dev, &evt);
+			break;
+		case NRF_GNSS_AGPS_DATA_ID:
+			LOG_DBG("A-GPS data needed message received");
```

Comment:
Rewrite to somthing "lighter"? I.e "A-GPS data update needed"

Reply:
Rewritten as suggested